File: Assignment2/app/src/main/java/edu/ucsc/cse118/assignment2/data/Workspace.kt
package edu.ucsc.cse118.assignment2.data

import kotlinx.serialization.Serializable

@Serializable
data class Workspace (
    val name: String,
    val channels: Array<Channel>,
)

@Serializable
data class Channel (
    val name: String,
    val messages: Array<Message>,
)

@Serializable
data class Message (
    val user: User,
    val date: String,
    val content: String,
)

@Serializable
data class User (
    val name: String,
    val email: String,
)

File: Assignment5/Assignment5/Data/Workspace.swift
//
//  Workspace.swift
//  Assignment5
//
//  Created by Kevin Crawford on 10/27/22.
//

import Foundation

struct Workspace: Identifiable, Decodable {
    let id: UUID
    let name: String
    var channels: [Channel]
}

extension Workspace {
  static let examples = [
    Workspace(id: UUID(), name: "Workspace", channels: [
      Channel(id: UUID(), name: "Channel", messages: [
        Message(id: UUID(), content: "Content", posted: "Date", member: Member(id: UUID(), name: "Name"))
      ])
    ])
  ]
}
